permissions/actions.js

```diff
@@ -0,0 +1,83 @@
+const diff = require('deep-diff');
+
+const CLOSE_CASE = 'closeCase';
+const REOPEN_CASE = 'reopenCase';
+const ADD_NOTE = 'addNote';
+const EDIT_NOTE = 'editNote';
+const ADD_REFERRAL = 'addReferral';
+const EDIT_REFERRAL = 'editReferral';
+const ADD_HOUSEHOLD = 'addHousehold';
+const EDIT_HOUSEHOLD = 'editHousehold';
+const ADD_PERPETRATOR = 'addPerpetrator';
+const EDIT_PERPETRATOR = 'editPerpetrator';
+const ADD_INCIDENT = 'addIncident';
+const EDIT_INCIDENT = 'editIncident';
+const EDIT_CASE_SUMMARY = 'editCaseSummary';
+
+const isPathEqual = (left, right) => left[0] === right[0] && left[1] === right[1];
+
+const isAddOrEditKind = kind => ['N', 'E', 'A'].includes(kind);
+
+const isEditCaseSummary = change =>
+  isAddOrEditKind(change.kind) && isPathEqual(change.path, ['info', 'summary']);
+
+const isCloseCase = change =>
+  change.kind === 'E' && isPathEqual(change.path, ['status']) && change.rhs === 'closed';
+
+const isReopenCase = change =>
+  change.kind === 'E' &&
+  isPathEqual(change.path, ['status']) &&
+  change.lhs === 'closed' &&
+  change.rhs === 'open';
+
+const isAddNote = change =>
+  isAddOrEditKind(change.kind) && isPathEqual(change.path, ['info', 'notes']);
+
+const isAddReferral = change =>
+  isAddOrEditKind(change.kind) && isPathEqual(change.path, ['info', 'referrals']);
+
+const isAddHousehold = change =>
+  isAddOrEditKind(change.kind) && isPathEqual(change.path, ['info', 'households']);
+
+const isAddPerpetrator = change =>
+  isAddOrEditKind(change.kind) && isPathEqual(change.path, ['info', 'perpetrators']);
+
+const isAddIncident = change =>
+  isAddOrEditKind(change.kind) && isPathEqual(change.path, ['info', 'incidents']);
+
+const getActions = (original, updated) => {
```

Comment:
Here it was also a decision to have the feature done in time. In the ideal scenario we would create different endpoints that do more specific things, and then update the front-end to call each of these endpoints, passing the values in the correct format, etc. It would take much longer.